src/dataframe/dataframe.jl

```diff
@@ -1508,3 +1508,63 @@ end
 function permutecols!(df::DataFrame, p::AbstractVector{Symbol})
     permutecols!(df, index(df)[p])
 end
+
+##############################################################################
+##
+## Expand & Complete dataframes
+##
+##############################################################################
+
+
+function expanddf(df::AbstractDataFrame, indexcols::Array{Symbol,1})
+    colnames = names(df)
```

Comment:
you can use `_names` to avoid a copy

Reply:
Or even avoid it and call `index(df)[indexcols]` and later work on integers - this will be more flexible.

Reply:
Sorry, could you explain a little more how that makes it more flexible? And what do you mean "and later work on integers"?

Reply:
You can supply to `index(df)[indexcols]` a wide selection of `indexcols` vales as opposed to your proposal. E.g. `indexcols` can be a `Regex`, or `Not` expression. Then `index(df)[indexcols]` guarantees you that you will receive as a result either `Int` or `Vector{Int}`. This way later you do not have to handle the multiple different possible options but work only with integers.